hibernate-core/src/main/java/org/hibernate/query/hql/internal/SemanticQueryBuilder.java

```diff
@@ -1498,12 +1500,24 @@ public SqmDynamicInstantiation<?> visitInstantiation(HqlParser.InstantiationCont
 		}
 
 		if ( !dynamicInstantiation.checkInstantiation( creationContext.getTypeConfiguration() ) ) {
-			final String typeName = dynamicInstantiation.getJavaType().getSimpleName();
+			final Class<?> type = dynamicInstantiation.getJavaType();
 			if ( dynamicInstantiation.isFullyAliased() ) {
-				throw new SemanticException( "Missing constructor or attributes for injection into type '" + typeName + "'", query );
+				throw new SemanticException("Missing constructor or attributes for injection into type '" + type + "'" +
```

Comment:
```suggestion
				throw new SemanticException("Missing constructor or attributes for injection into type '" + type.getSimpleName() + "'" +
```

Reply:
I think a type description like `com.excemple.A` would be clearer than just
showing `A`, especially if there are classes with duplicate names, and
would make it easier to find problematic classes more quickly.

Reply:
I'm not certain I agree, but anyway that's not what you have there. You are calling `Class.toString()`, does not just return the qualified name.

Reply:
I understand your concern, I've used `Class.getName()` replace `Class.toString()`